tests/integration/backward_compatible/proxy/map_test.py

```diff
@@ -790,32 +793,64 @@ def test_int_average_with_attribute_path(self):
+        average = self.map.aggregate(int_avg(), greater_or_equal("this", 47))
         self.assertEqual(48, average)
 
     def test_int_sum(self):
-        sum = self.map.aggregate(int_sum())
-        self.assertEqual(1225, sum)
+        sum_ = self.map.aggregate(int_sum())
+        self.assertEqual(1225, sum_)
 
     def test_int_sum_with_attribute_path(self):
-        sum = self.map.aggregate(int_sum("this"))
-        self.assertEqual(1225, sum)
+        sum_ = self.map.aggregate(int_sum("this"))
+        self.assertEqual(1225, sum_)
 
     def test_int_sum_with_predicate(self):
-        sum = self.map.aggregate(int_sum(), predicate=greater_or_equal("this", 47))
-        self.assertEqual(144, sum)
+        sum_ = self.map.aggregate(int_sum(), greater_or_equal("this", 47))
+        self.assertEqual(144, sum_)
 
     def test_fixed_point_sum(self):
-        sum = self.map.aggregate(fixed_point_sum())
-        self.assertEqual(1225, sum)
+        sum_ = self.map.aggregate(fixed_point_sum())
+        self.assertEqual(1225, sum_)
 
     def test_fixed_point_sum_with_attribute_path(self):
-        sum = self.map.aggregate(fixed_point_sum("this"))
-        self.assertEqual(1225, sum)
+        sum_ = self.map.aggregate(fixed_point_sum("this"))
+        self.assertEqual(1225, sum_)
 
     def test_fixed_point_sum_with_predicate(self):
-        sum = self.map.aggregate(fixed_point_sum(), predicate=greater_or_equal("this", 47))
-        self.assertEqual(144, sum)
+        sum_ = self.map.aggregate(fixed_point_sum(), greater_or_equal("this", 47))
+        self.assertEqual(144, sum_)
+
+    def test_distinct(self):
+        distinct_values = self.map.aggregate(distinct())
+        self.assertEqual(set(range(50)), distinct_values)
+
+    def test_distinct_with_attribute_path(self):
+        distinct_values = self.map.aggregate(distinct("this"))
+        self.assertEqual(set(range(50)), distinct_values)
+
+    def test_distinct_with_predicate(self):
+        distinct_values = self.map.aggregate(distinct(), greater_or_equal("this", 10))
+        self.assertEqual(set(range(10, 50)), distinct_values)
+
+    def test_max_by(self):
+        max_item = self.map.aggregate(max_by("this"))
+        self.assertEqual("key-49", max_item.key)
+        self.assertEqual(49, max_item.value)
+
+    def test_max_by_with_predicate(self):
+        max_item = self.map.aggregate(max_by("this"), less_or_equal("this", 10))
+        self.assertEqual("key-10", max_item.key)
+        self.assertEqual(10, max_item.value)
+
+    def test_min_by(self):
+        min_item = self.map.aggregate(min_by("this"))
+        self.assertEqual("key-0", min_item.key)
+        self.assertEqual(0, min_item.value)
+
+    def test_min_by_with_predicate(self):
+        min_item = self.map.aggregate(min_by("this"), greater_or_equal("this", 10))
+        self.assertEqual("key-10", min_item.key)
+        self.assertEqual(10, min_item.value)
```

Comment:
We don't need to add test all of long, double, and int, that would be unnecessary. However, if we can test `distinct` with a map that includes same values, it would be better. Right now, the values are from 0..49, which are unique.

Reply:
Improved the tests so that we have some duplicate values

Reply:
changes look good, thanks